docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+type.
+
+
+Core data types
+---------------
+
+.. list-table:: Data types
+   :header-rows: 1
+
+   * - Identifier
+     - Numerical type
+     - Size (no. bytes)
+     - Byte order
+   * - ``bool``
+     - Boolean, with False encoded as ``\\x00`` and True encoded as ``\\x01``
+     - 1
+     - None
+   * - ``i1``
+     - signed integer
+     - 1
+     - None
+   * - ``<i2``
+     - signed integer
+     - 2
+     - little-endian
+   * - ``<i4``
+     - signed integer
+     - 4
+     - little-endian
+   * - ``<i8``
+     - signed integer
+     - 8
+     - little-endian
+   * - ``>i2``
+     - signed integer
+     - 2
+     - big-endian
+   * - ``>i4``
+     - signed integer
+     - 4
+     - big-endian
+   * - ``>i8``
+     - signed integer
+     - 8
+     - big-endian
+   * - ``u1``
+     - unsigned integer
+     - 1
+     - None
+   * - ``<u2``
+     - unsigned integer
+     - 2
+     - little-endian
+   * - ``<u4``
+     - unsigned integer
+     - 4
+     - little-endian
+   * - ``<u8``
+     - unsigned integer
+     - 8
+     - little-endian
+   * - ``>u2``
+     - unsigned integer
+     - 2
+     - big-endian
+   * - ``>u4``
+     - unsigned integer
+     - 4
+     - big-endian
+   * - ``>u8``
+     - unsigned integer
+     - 8
+     - big-endian
+   * - ``<f2``
+     - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
+     - 2
+     - little-endian
+   * - ``<f4``
+     - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
+     - 4
+     - little-endian
+   * - ``<f8``
+     - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
+     - 8
+     - little-endian
+   * - ``>f2``
+     - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
+     - 2
+     - big-endian
+   * - ``>f4``
+     - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
+     - 4
+     - big-endian
+   * - ``>f8``
+     - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
+     - 8
+     - big-endian
+   * - ``r*`` (Optional)
+     - raw bits,  use for extension type fallbacks
+     - variable, given by ``*``, is limited to be a multiple of 8.
+     - N/A
+
+
+Floating point types correspond to basic binary interchange formats as
+defined by IEEE 754-2008.
+
+Additionally to these base types, an implementation should also handle the
+raw/opaque pass-through type designated by the lower-case letter ``r`` followed
+by the number of bits, multiple of 8. For example, ``r8``, ``r16``, and ``r24``
+should be understood as fall-back types of respectively 1, 2, and 3 byte length.
+
+Zarr v3 is limited to type sizes that are a multiple of 8 bits but may support
+other type sizes in later versions of this specification.
+
+
+.. note::
+
+    We are explicitly looking for more feedback and prototypes of code using the ``r*``,
+    raw bits, for various endianness and whether the spec could be made clearer.
+
+.. note::
```

Comment:
Since this is a very important feature: are there any plans to add this as an extension (or multiple extensions since there are different ways to encode varlen / ragged arrays)? If yes, it could help to link to some efforts here.